src/tests/sqlsmith/src/bin/main.rs

```diff
@@ -0,0 +1,104 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use clap::Parser as ClapParser;
+use risingwave_sqlparser::ast::Statement;
+use risingwave_sqlparser::parser::Parser;
+use risingwave_sqlsmith::{sql_gen, Table};
+use tokio_postgres::NoTls;
+
+#[derive(ClapParser, Debug, Clone)]
+#[clap(about, version, author)]
+struct Opt {
+    /// The database server host.
+    #[clap(short, long, default_value = "localhost")]
+    host: String,
+
+    /// The database server port.
+    #[clap(short, long, default_value = "4566")]
+    port: u16,
+
+    /// The database name to connect.
+    #[clap(short, long, default_value = "dev")]
+    db: String,
+
+    /// The database username.
+    #[clap(short, long, default_value = "root")]
+    user: String,
+
+    /// The database password.
+    #[clap(short = 'w', long, default_value = "")]
+    pass: String,
+
+    /// Path to the testing data files.
+    #[clap(short, long)]
+    testdata: String,
+}
+
+async fn create_tables(opt: &Opt, client: &tokio_postgres::Client) -> Vec<Table> {
+    log::info!("Preparing tables...");
+
+    let sql = std::fs::read_to_string(format!("{}/tpch.sql", opt.testdata)).unwrap();
+
+    let statements =
+        Parser::parse_sql(&sql).unwrap_or_else(|_| panic!("Failed to parse SQL: {}", sql));
+    for stmt in statements.iter() {
+        let create_sql = format!("{}", stmt);
+        client.execute(&create_sql, &[]).await.unwrap();
+    }
+    statements
+        .into_iter()
+        .map(|s| match s {
+            Statement::CreateTable { name, columns, .. } => Table {
+                name: name.0[0].value.clone(),
+                columns: columns.iter().map(|c| c.clone().into()).collect(),
+            },
+            _ => panic!("Unexpected statement: {}", s),
+        })
+        .collect()
+}
+
+#[tokio::main(flavor = "multi_thread", worker_threads = 5)]
+async fn main() {
+    env_logger::init();
+
+    let opt = Opt::parse();
+    let (client, connection) = tokio_postgres::Config::new()
+        .host(&opt.host)
+        .port(opt.port)
+        .dbname(&opt.db)
+        .user(&opt.user)
+        .password(&opt.pass)
+        .connect_timeout(Duration::from_secs(5))
+        .connect(NoTls)
+        .await
+        .unwrap_or_else(|e| panic!("Failed to connect to database: {}", e));
+    tokio::spawn(async move {
+        if let Err(e) = connection.await {
+            log::error!("Postgres connection error: {:?}", e);
+        }
+    });
+
+    let tables = create_tables(&opt, &client).await;
+
+    let mut rng = rand::thread_rng();
```

Comment:
It seems that `ThreadRng` is seeded from `OsRng`, which means that the seed is different for each test. I would suggest to use a fixed seed so that we can reproduce each test.

Reply:
I think the typical need for reproducibility is to reproduce the same bug given the same SQL. Hence whether the SQL generator uses a fixed seed is not important.